yt/funcs.py

```diff
@@ -1086,12 +1086,20 @@ def array_like_field(data, x, field):
         return data.ds.quan(x, units)
 
 
+def _full_type_name(obj: object = None, /, *, cls: Optional[type] = None) -> str:
+    if cls is not None and obj is not None:
+        raise TypeError("_full_type_name takes an object or a class, but not both")
+    if cls is None:
+        cls = obj.__class__
+    prefix = f"{cls.__module__}." if cls.__module__ != "builtins" else ""
+    return f"{prefix}{cls.__name__}"
+
+
 def validate_3d_array(obj):
     if not is_sequence(obj) or len(obj) != 3:
         raise TypeError(
-            "Expected an array of size (3,), received '{}' of length {}".format(
-                str(type(obj)).split("'")[1], len(obj)
-            )
+            f"Expected an array of size (3,), "
+            f"received {_full_type_name(obj)!r} of length {len(obj)}"
```

Comment:
Why `_full_type_name(obj)!r` rather than just `_full_type_name(obj)`? Is it just to get the `''` for free?

Reply:
> s it just to get the '' for free?

yes